### PR TITLE
[openssl] no_module option for OpenSSL v3.0.x legacy providers

### PR DESCRIPTION
**Pull request by jngrb**
Specify library name and version:  **openssl/3.0.x**

reason: legacy providers are compiled into shared modules
unless no_module/no-module is configured.
Set `no_module=True` to have a fully statically
linked package without any external dependencies.

References:

* <https://mta.openssl.org/pipermail/openssl-users/2022-February/014904.html>

Added legacy provider use cases to the test package
(if enabled)

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by jngrb** [comment]
Hint: if the no_module option is not set, consumers of the pre-built seemingly 'statically linked' packages ( _shared=False_ ) might not be able to load the legacy provider from the legacy module that is a shared object unless the no_module option is set.

Compare #10409

**Review by prince-chrismc — APPROVED** [review]
Thank you for links pointing everything out ❤️

File: recipes/openssl/3.x.x/conanfile.py [hunk 1]
```diff
@@ -55,6 +56,7 @@ class OpenSSLConan(ConanFile):
         "no_md2": [True, False],
         "no_md4": [True, False],
         "no_mdc2": [True, False],
+        "no_module": [True, False],
```
Comment (jgsogo) [inline recipes/openssl/3.x.x/conanfile.py:59]: How is this option used in the recipe itself? I would expect something else to change in this `conanfile.py`

**Review by jgsogo — CHANGES_REQUESTED** [review]

File: recipes/openssl/3.x.x/conanfile.py (hunk 1 above)
Comment (jngrb) [reply recipes/openssl/3.x.x/conanfile.py:59]: > How is this option used in the recipe itself?

All options that don't need special treatment are handled in the _for_ loop in lines 438/440 of the method `_configure_args` in openssl's _conanfile.py_

In this case, the option `openssl:no_module=True` would be translated into an argument _no-module_ for the configure call

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

File: recipes/openssl/3.x.x/conanfile.py (hunk 1 above)
Comment (jgsogo) [reply recipes/openssl/3.x.x/conanfile.py:59]: Thanks for the clarification

**Pull request merged**